packages/dd-trace/test/git_metadata.spec.js

```diff
@@ -0,0 +1,155 @@
+'use strict'
+const nock = require('nock')
+const os = require('os')
+const fs = require('fs')
+
+describe.only('git_metadata', () => {
+  let gitMetadata
+
+  const latestCommits = ['87ce64f636853fbebc05edfcefe9cccc28a7968b', 'cc424c261da5e261b76d982d5d361a023556e2aa']
+  const temporaryPackFile = `${os.tmpdir()}/1111-87ce64f636853fbebc05edfcefe9cccc28a7968b.pack`
+  const secondTemporaryPackFile = `${os.tmpdir()}/1111-cc424c261da5e261b76d982d5d361a023556e2aa.pack`
+
+  let getLatestCommitsStub
+  let getRepositoryUrlStub
+  let getCommitsToUploadStub
+  let generatePackFilesForCommitsStub
+
+  before(() => {
+    process.env.DD_API_KEY = 'api-key'
+    fs.writeFileSync(temporaryPackFile, '')
+    fs.writeFileSync(secondTemporaryPackFile, '')
+  })
+
+  after(() => {
+    delete process.env.DD_API_KEY
+    fs.unlinkSync(temporaryPackFile)
+    fs.unlinkSync(secondTemporaryPackFile)
+  })
+
+  beforeEach(() => {
+    getLatestCommitsStub = sinon.stub().returns(latestCommits)
+    getCommitsToUploadStub = sinon.stub().returns(latestCommits)
+    getRepositoryUrlStub = sinon.stub().returns('git@github.com:DataDog/dd-trace-js.git')
+
+    generatePackFilesForCommitsStub = sinon.stub().returns([temporaryPackFile])
+
+    gitMetadata = proxyquire('../src/git_metadata', {
```

Comment:
With the core rewrite coming up, can this test be written as an integration test instead of a unit test with mocks? It's likely that things will move around and unit tests will break.

Reply:
I'll see what I can do 👍